Current_SamplingFunctionality.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <cstring>
+#include "Current_SamplingFunctionality.hpp"
+using namespace std;
+
+
+unsigned int ConsecutiveRanges(int Current_Samples[], int Current_Samples_size) {
+
+    unsigned int count = 1;
+    int idx; // index
+    // check if the current element & previous element step is 1.
+    for(idx=0; idx < Current_Samples_size; idx++) {
+
+        if ((Current_Samples[idx + 1] - Current_Samples[idx]) == 1 ) {
+            count = count + 1;
+        }
+    }
+    return count;
+}
+
+bool SampleRangeReadings(int Current_Samples[], int Current_Samples_size, int ReadingsCount)
```

Comment:
Looks like this function is only needed for testing? If so, move it to `Test_Cases_main.cpp`
Don't mix test and production code.